src/core/Database.h

```diff
@@ -63,6 +63,13 @@ class Database : public ModifiableObject
     };
     static const quint32 CompressionAlgorithmMax = CompressionGZip;
 
+    enum SaveAction
+    {
+        Atomic, // Saves are transactional and atomic
+        NonAtomic, // Save operation may be interrupted
```

Comment:
Still not sure if this is the correct name and description. This IS atomic after all, as long as everything happens within the same file system. If tmp is on another file system, though, it basically turns into `DirectWrite`.